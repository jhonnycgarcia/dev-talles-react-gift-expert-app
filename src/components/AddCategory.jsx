import PropTypes from 'prop-types';

import { useState } from "react"

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    };

    const onSubmit = (e) => {
        e.preventDefault();
        const newValue = inputValue.trim();
        if(newValue.length <= 1) { return; }
        console.log('dentro', newValue);
        onNewCategory(newValue);
        setInputValue('');
    };

    return (
        <form onSubmit={onSubmit} aria-label='form'>
            <input
                type="text"
                placeholder="Buscar Gifts"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    );
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired,
};
