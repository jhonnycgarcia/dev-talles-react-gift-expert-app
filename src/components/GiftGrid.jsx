import PropTypes from 'prop-types';

import { GiftItem } from './GiftItem';

import { useFetchGifs } from '../hooks/useFetchGifs';

export const GiftGrid = ({ category }) => {

    const { images, isLoading} = useFetchGifs(category);

    return (
        <>
            <h3 aria-label='category-title'>{category}</h3>

            { isLoading && <h2>Cargando...</h2> }

            <div className='card-grid'>
                { images.map((image) => (
                    <GiftItem 
                        key={image.id} 
                        {...image}
                    />
                )) }
            </div>
        </>
    );
}

GiftGrid.propTypes = {
    category: PropTypes.string.isRequired,
};
