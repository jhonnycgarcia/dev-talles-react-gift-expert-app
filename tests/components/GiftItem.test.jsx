import { render } from "@testing-library/react";
import { GiftItem } from "../../src/components/GiftItem";


describe('Pruebas en <GiftItem />', () => {

    const title = 'One Punch Man GIF';
    const url = 'https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=e93ddaacz1584r9i1wmkmnpu497c8362x8lpx0ldukkmvx9h&ep=v1_gifs_search&rid=giphy.gif&ct=g';

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GiftItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

});