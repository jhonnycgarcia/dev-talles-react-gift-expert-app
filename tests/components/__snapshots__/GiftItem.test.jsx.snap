// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="One Punch Man GIF"
      src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=e93ddaacz1584r9i1wmkmnpu497c8362x8lpx0ldukkmvx9h&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      One Punch Man GIF
    </p>
  </div>
</div>
`;
